book/src/rfcs/keccak.md

```diff
@@ -32,3 +42,126 @@ Before we describe the specific gate design approaches, below are some constrain
+y &= 26, 10, 26, 10, 26, 10...
+\end{align*}
+$$
+
+Note that our valid witness is part of that set of solutions, meaning $x=1$ and $y=10$. Of course, we can also find another dishonest instantiation such as $x=0$ and $y=26$. Perhaps one could think that we do not need to worry about this case, because the resulting rotation word would be $0+26=26$, and if we later use that result as an input to a subsequent gate such as XOR, the value $26$ would not fit and at some point the constraint system would complain. Nonetheless, we still have other solutions to worry about, such as $(x=3, y=10)$ or $(x=5, y=10)$, since they would result in a rotated word that would fit in the word length of 4 bits, yet would be incorrect (not equal to $11$). 
+
+All of the above incorrect solutions differ in one thing: they have different bit lengths. This means that we need to range check the values for the excess and shifted witnesses to make sure they have the correct length. 
+
+### Sufficiency of range checks
+
+In the following, we will give a proof that performing range checks for these values is not only necessary but also sufficient to prove that the rotation gate is sound. Meaning, we will prove that there are no two possible solutions of the decomposition constraint that while having correct bit lengths. Now, for the sake of robustness, we will consider 64-bit words and fields with at least twice the bit length of the words (as is our case). 
+
+We will proceed by __contradiction__. Suppose there are two different solutions to the following diophantic equation:
+
+$$
+\begin{align*}
+\forall k \in \mathbb{F}_n: & \\
+x &= k \\
+y &= w \cdot 2^r - 2^{64} \cdot k
+\end{align*}
+$$
+
+where $k$ is a parameter to instantiate the solutions, $w$ is the word to be rotated, $r$ is the rotation amount, and $n$ is the field length.
+
+Then, that means that there are two different solutions, $(x=a<2^r, y=b<2^{64})$ and $(x=a'<2^r, y=b'<2^{64})$ with at least $a \neq a'$ or $b \neq b'$. We will show that this is impossible.
+
+If both are solutions to the same equation, then: 
+$$
+\begin{align*}
+w \cdot 2^r &= a \cdot 2^{64} + b \\
+w \cdot 2^r &= a'\cdot 2^{64} + b'
+\end{align*}
+$$
+means that $a \cdot 2^{64} + b = a'\cdot 2^{64} + b'$. Moving terms to the left side, we have an equivalent equality: $2^{64}(a-a') + (b-b')=0 \text{ mod } n$. There are three cases to consider:
+
+- $a = a'$ and $b \neq b'$: then $(b - b') \equiv_n 0$ and this can only happen if $b' = b + n$. But since $n > 2^{64}$, then $b'$ cannot be smaller than $2^{64}$ as it was assumed. CONTRADICTION.
+  
+- $b = b'$ and $a \neq a'$: then $2^{64}(a - a') \equiv_n 0$ and this can only happen if $a' = a + n$. But since $n > 2^r$, then $a'$ cannot be smaller than $2^r$ as it was assumed. CONTRADICTION.
+
+- $a\neq a'$ and $b \neq b'$: then we have something like $2^{64} \alpha + \beta \equiv_n 0$. We choose to express $\alpha$:
+  - $$2^{64} \alpha = n - \beta = n - b + b' > n - 2^{64} + 1$$
```

Comment:
Something's up with the bullets here too.  Maybe either remove bullets or wrap equations with single `$` instead

Reply:
it compiles nice with mdbook though